org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/ex/ExternalAntlrLexerFragment.java

```diff
@@ -0,0 +1,363 @@
+					StringConcatenationClient binding = new StringConcatenationClient() {
+						@Override
+						protected void appendTo(TargetStringConcatenation target) {
+							target.append("binder.bind(org.eclipse.xtext.ide.editor.contentassist.antlr.internal.Lexer.class)"+
+									".annotatedWith(com.google.inject.name.Names.named(" +
+									"org.eclipse.xtext.ide.LexerIdeBindings.CONTENT_ASSIST" +
+									")).to(" + lexerGrammar +".class);");
+						}
+					};
+					new GuiceModuleAccess.BindingFactory().addConfiguredBinding("ContentAssistLexer", binding).contributeTo(getLanguage().getIdeGenModule());
+			
+				} 
+				if (this.getProjectConfig().getEclipsePlugin().getRoot() != null) {
+					StringConcatenationClient binding = new StringConcatenationClient() {
+						@Override
+						protected void appendTo(TargetStringConcatenation target) {
+							target.append("binder.bind(org.eclipse.xtext.ide.editor.contentassist.antlr.internal.Lexer.class)"+
+									".annotatedWith(com.google.inject.name.Names.named(" +
+									"org.eclipse.xtext.ide.LexerIdeBindings.CONTENT_ASSIST" +
+									")).to(" + lexerGrammar +".class);");
+						}
+					};
+					new GuiceModuleAccess.BindingFactory().addConfiguredBinding("ContentAssistLexer", binding).contributeTo(getLanguage().getEclipsePluginGenModule());
+				}
+			}
+			
+			
+		
+		
+		IXtextGeneratorFileSystemAccess srcGenFsa = this.getProjectConfig().getRuntime().getSrcGen();
+		IXtextGeneratorFileSystemAccess srcFsa = this.getProjectConfig().getRuntime().getSrc();
+		if (contentAssist || highlighting) {
+			if (this.getProjectConfig().getGenericIde().getRoot() != null) {
+				srcGenFsa = this.getProjectConfig().getGenericIde().getSrcGen();
+				srcFsa = this.getProjectConfig().getGenericIde().getSrc();
+			} else {
+				srcGenFsa = this.getProjectConfig().getEclipsePlugin().getSrcGen();
+				srcFsa = this.getProjectConfig().getEclipsePlugin().getSrc();
+			}
+		}
+		String srcGenPath = srcGenFsa.getPath();
+		String srcPath = srcFsa.getPath();
+		String grammarFile = srcPath + "/" + getLexerGrammar().replace('.', '/') + ".g";
+		String generateTo = "";
+		if (lexerGrammar.lastIndexOf('.') != -1) {
+			generateTo = lexerGrammar.substring(0, lexerGrammar.lastIndexOf('.'));
+		}
+		generateTo = srcGenPath + "/" + generateTo.replace('.', '/');
+		addAntlrParam("-fo");
+		addAntlrParam(generateTo);
+		final String encoding = codeConfig.getEncoding();
+		getAntlrTool().runWithEncodingAndParams(grammarFile, encoding, getAntlrParams());
+		Charset charset = Charset.forName(encoding);
+		String javaFile = srcGenPath+"/"+getLexerGrammar().replace('.', '/')+".java";
+		splitLexerIfEnabled(javaFile, charset);
+		suppressWarningsImpl(javaFile, charset);
+		normalizeLineDelimiters(javaFile, charset);
+		normalizeTokens(javaFile, charset);
+	}
+	
+	/**
+	 * @since 2.9
+	 */
+	protected void splitLexerIfEnabled(String lexerJavaFile, Charset encoding) {
```

Comment:
Is it worthwhile to reuse stuff from the abstract Antlr fragments?

Reply:
i dont have any overview there